docs/references/security/rust-canister-development-security-best-practices.md

```diff
@@ -202,27 +202,29 @@ For example, consider the following Motoko code:
 
 The first message that is executed here are lines 2-3, until the inter-canister call is made using the `await` syntax (orange box). The second message executes lines 3-5: when the inter-canister call returns (blue box). We call this part the _callback_ of the inter-canister call. The two messages involved in this example will always be scheduled sequentially. 
 
-**Property 3**: Messages from interleaving calls have no reliable execution ordering.
+**Property 3**: Successfully delivered requests are received in the order in which they were sent. In particular, if a canister A sends `m1` and `m2` to canister B in that order, then, if both are accepted, `m1` is executed before `m2`. 
+
+**Property 4**: Messages from interleaving calls have no reliable execution ordering.
 
 Let's consider the above example code again, and assume the method `example` is called twice in parallel, the resulting calls being Call 1 and Call 2. The following illustration shows two possible message orderings. On the left, the first call's messages are scheduled first, and only then the second call's messages are executed. On the right, we see another possible message scheduling, where the first messages of each call are executed first. Your code should result in a correct state regardless of the message ordering.
```

Comment:
It would be nice if this bit would reference the property 3. I.e. something like "despite messages being received in order on the target canister (property 3) no assumptions can be made about the interleavings of messages from different callers".

Reply:
Good point. I added this aspect but chose somewhat different wording compared to what you are suggesting. I hope that makes it clearer. PTAL.